packages/theme-check-common/src/JSONValidator.ts

```diff
@@ -0,0 +1,52 @@
+import { LanguageService, TextDocument, getLanguageService } from 'vscode-json-languageservice';
+import { JsonValidationSet, SourceCodeType, ValidateJSON } from './types';
+
+/** This simply memoizes the result to prevent over fetching */
+export class JSONValidator {
+  private service: LanguageService;
+
+  constructor(private jsonValidationSet: JsonValidationSet) {
+    this.jsonValidationSet = jsonValidationSet;
+    this.service = getLanguageService({
+      schemaRequestService: this.getSchemaForURI.bind(this),
+    });
+    this.service.configure({
+      schemas: Object.values(this.jsonValidationSet.schemas).map((schemaDefinition) => ({
+        uri: schemaDefinition.uri,
+        fileMatch: schemaDefinition.fileMatch,
+      })),
+    });
+  }
+
+  /**
+   * Will return an array of diagnostics for the given source code and JSON string.
+   *
+   * It's up to the caller to determine where in the file those should be.
+   * (presumably by doing some offset logic)
+   */
+  public validate: ValidateJSON<SourceCodeType> = async (sourceCode, jsonString) => {
+    const jsonTextDocument = TextDocument.create(
+      'file:' + sourceCode.absolutePath,
+      'json',
+      0,
+      jsonString,
+    );
+    const jsonDocument = this.service.parseJSONDocument(jsonTextDocument);
```

Comment:
This here is kind of subtle. For one, there is no document manager in theme-check. We only have an array of SourceCode to deal with. 

We _could_ add those to the interface of SourceCode, but I think it would be rather weird. For one, in Liquid files, this only gets called with the body of `{% schema %}` tags. In JSON files, we use the contents as is. Even then, there's quite a bit of subtlety about offsetting the startIndex with the endIndex of the {% schema %} blockStart. 

I'm not 100% sure how we would put that on the SourceCode interface tbh. Right now it's this:
```ts
export type SourceCode<T = SourceCodeType> = T extends SourceCodeType
  ? {
      /** A normalized absolute path to the file. Assumes forwards slashes. */
      absolutePath: string;
      /** The type is used as a discriminant for type narrowing */
      type: T;
      /** The version is used by the Language Server to make sure the Client and Server are in sync */
      version?: number;
      /** The contents of the file */
      source: string;
      /** The AST representation of the file, or an Error instance when the file is unparseable */
      ast: AST[T] | Error;
    }
  : never;
```

If we were to preparse section files, we'd need a second ast attribute in there. But we'd need to have that attribute for all files, even non-section files.